contribute/design/1.x/cstor/cstor-status.md

```diff
@@ -0,0 +1,220 @@
+
+## Motivation
+
+The current CStor CRs don't provide appropriate information about the state of the CR and the reason for that state.
+
+### Goals
+  - Make CSPC and CSPI status informative enough to know the current state of the CR and the reason for the given state.
+
+## Proposal
+
+### User Stories
+
+As a OpenEBS user, the information regarding a CR should be present in the status or events of the CR itself rather than looking for it in controller logs.
+
+### Proposed Implementation
+
+Kuberentes provides a framework for status conditions and events which can be used to pass relevant information from the controllers to the CRs. 
+
+#### CSPI Status
+
+The current CSPI status gives us the following information via kubectl get command which are the zpool properties taken from the zfs commands and parsed to cspi status.
+
+```diff
+mayadata:maya$ kubectl -n openebs get cspi
+-  NAME                  HOSTNAME    ALLOCATED   FREE    CAPACITY  STATUS   AGE
+-  sparse-pool-1-7xq5    127.0.0.1     218K      9.94G    9.94G    ONLINE   134m
++  NAME                  HOSTNAME    ALLOCATED   FREE    CAPACITY  HEALTHYVOLUMES  STATUS   AGE
++  sparse-pool-1-7xq5    127.0.0.1     218K      9.94G    9.94G         3/5        ONLINE   134m
+```
+
+As the CSPI and CVR have their controllers clubbed together as cspi-mgmt having CVR information on CSPI will help with the debugging process.
+The current CSPI gives the status as a form of Status.Phase which is the zpool state representation. This Phase is updated by the CSPI mgmt container. The mgmt container gets the health status from the zpool command and sets that to the phase of CSPI. The possible CSPI phases are:
+
+**Online** : The device or virtual device is in normal working order. Although some transient errors might still occur, the device is otherwise in working order.
+
+**Degraded** : The virtual device has experienced a failure but can still function. This state is most common when a mirror or RAID-Z device has lost one or more constituent devices. The fault tolerance of the pool might be compromised, as a subsequent fault in another device might be unrecoverable.
+
+**Faulted** : The device or virtual device is completely inaccessible. This status typically indicates the total failure of the device, such that ZFS is incapable of sending data to it or receiving data from it. If a top-level virtual device is in this state, then the pool is completely inaccessible.
+
+**Offline** : The device has been explicitly taken offline by the cspi-mgmt controller.
+
+**Unavail** : The device or virtual device cannot be opened. In some cases, pools with UNAVAIL devices appear in DEGRADED mode. If a top-level virtual device is UNAVAIL, then nothing in the pool can be accessed.
+
+**Removed** : The device was physically removed while the system was running. Device removal detection is hardware-dependent and might not be supported on all platforms.
+
```

Comment:
The phases above -- are those the exhaustive list of phases for a store pool? 
We can have phases for cstor pool from data plane side as well as control plane.
@vishnuitta  to confirm on the exhaustive list. 

From the control plane side -- the phases can be e.g. `Init` or `Unknown` 

Listing phases for cspi from both control plane + data plane would set the boundaries more clear...

Reply:
states of pool in current implementation is exhaustive list.